Add a separate media type for 4k/UHD

gjnoonan opened the pull request:
As requested on the forums, as well as scratching my own itch. 4K releases are now treated as separate media-type when imported. This allows the user to not only; see visually which releases are 4k,if they have format icons enabled within their config, but also filter to 4K releases.

Works nicely on a test DB I put together.

![Screenshot 2020-01-05 at 06 50 19](https://user-images.githubusercontent.com/702/71776330-a5faf800-2f87-11ea-842a-f1c634f571ff.png)

![Screenshot 2020-01-05 at 06 51 00](https://user-images.githubusercontent.com/702/71776336-bdd27c00-2f87-11ea-924e-e051ad9add98.png)

![Screenshot 2020-01-05 at 06 51 29](https://user-images.githubusercontent.com/702/71776337-ce82f200-2f87-11ea-9c2e-23f041d28bdc.png)

Still left to do is the movie picker/chooser, but this won't be too much work for a later commit.

_More information in the commit messages, I would suggest not squashing commits when you merge._

gjnoonan committed:
mediatypes: add ultrahd for 4K releases

This is some prep work for later commits in this series, adding some new
media types which will  allow us to handle 4K releases separately from
Blu-Ray.

Within the DVDProfiler export there are 4 media types exported; DVD,
HDDVD, BluRay, and  UltraHD. As no releases exist which contain a UHD
disc along with a HDDVD, we are left with the following possible
combinations:

  * ULTRAHD - a release containing just a UHD disc
  * ULTRAHD_BLURAY - a release with both a UHD disc and a Blu-ray
  * ULTRAHD_BLURAY_DVD - a release with UHD, Blu-ray and DVD discs

Some initial graphics are included, but these are expected to change.

Signed-off-by: Gavin-John Noonan <mail@gjnoonan.co.uk>

gjnoonan committed:
i18n: add translation for ultrahd

In 431068f039bf (mediatypes: add ultrahd for 4K releases, 2020-01-05) we
introduced a new media type for ultrahd/4k releases. Add a translation
for it in the language files.

Left untranslated is russian.

Signed-off-by: Gavin-John Noonan <mail@gjnoonan.co.uk>

gjnoonan committed:
import: teach the import about ultrahd/4k media

When performing a collection update, classify a DVD as a ultrahd
release based on the media types set up in 431068f039bf (mediatypes: add
ultrahd for 4K releases, 2020-01-05).

UHD releases generally have the same video formats as Blu-ray and
HD-DVD, so we can re-use them.

Signed-off-by: Gavin-John Noonan <mail@gjnoonan.co.uk>

gjnoonan committed:
frontend: display format icons for ultrahd

Now that 4k/UHD titles are classified distinctly from standard Blu-ray
releases, display the correct images when viewing a 4k/UHD release.

Signed-off-by: Gavin-John Noonan <mail@gjnoonan.co.uk>

gjnoonan committed:
search: teach search how to filter by uhd releases

The ability was already there to filter searches by media type, so allow
a user to filter by the ULTRAHD media types too.

Signed-off-by: Gavin-John Noonan <mail@gjnoonan.co.uk>

gjnoonan committed:
popup: display the right term for ultrahd titles

Signed-off-by: Gavin-John Noonan <mail@gjnoonan.co.uk>

gjnoonan committed:
smallshow/mobile: display ultrahd media type:

Signed-off-by: Gavin-John Noonan <mail@gjnoonan.co.uk>

julmud commented:
Wow, thanks a lot for this Pull Request! I'll have a look at it and merge it as soon as I can find the time to do it, but it probably won't be today nor tomorrow, alas.

gjnoonan commented:
No worries! I have a few more lined up to send in once I finish testing them, and splitting the hunks--but none are dependent on this being in master :-)

julmud commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+/vendor/
+ERROR_LOG
```
Thanks. I wasn't sure whether or not it was a good idea to commit the file, but it makes sense to ignore it at least. We should probably also add the DEBUG_LOG file to the ignore list. That one should never be committed, but it can be created when developers are trying to understand what happens in the code. ;-)
```suggestion
ERROR_LOG
DEBUG_LOG
```

gjnoonan replied:
I have now pulled this out into it's own commit 053d2973887aeb406fef28740da4ba674cc1864b

_edit: the commit hash changed, as I edited the commit message and pushed it back up_

julmud commented on globalinits.php:
```diff
@@ -63,7 +66,11 @@
-	MEDIA_TYPE_BLURAY_DVD	=> array('FormatIcon' => 'gfx/iconbluraypng.png', 'Icon' => 'gfx/bluray.jpg', 'Banner' => 'gfx/Banner_BluRayDVD.png')
+	MEDIA_TYPE_BLURAY_DVD	=> array('FormatIcon' => 'gfx/iconbluraypng.png', 'Icon' => 'gfx/bluray.jpg', 'Banner' => 'gfx/Banner_BluRayDVD.png'),
+	MEDIA_TYPE_ULTRAHD	=> array('FormatIcon' => 'gfx/iconultrahd.png', 'Icon' => 'gfx/ultrahd.jpg', 'Banner' => 'gfx/Banner_UltraHD.png'),
+	MEDIA_TYPE_ULTRAHD_BLURAY	=> array('FormatIcon' => 'gfx/iconultrahd.png', 'Icon' => 'gfx/ultrahd.jpg', 'Banner' => 'gfx/Banner_UltraHDBD.png'),
```
Small alignment problem
```suggestion
	MEDIA_TYPE_ULTRAHD_BLURAY    => array('FormatIcon' => 'gfx/iconultrahd.png', 'Icon' => 'gfx/ultrahd.jpg', 'Banner' => 'gfx/Banner_UltraHDBD.png'),
```

gjnoonan replied:
Good spot thanks, I thought I caught them all when I went back through.

gjnoonan replied:
I have not resolved this on with the suggestion, as it still does not look right. The array keys have become too long I think. I will push a patch with a larger change clearing up the array itself.

julmud commented on incupdate.php:
```diff
@@ -476,12 +476,20 @@ function InitialiseCommonTable($which) {
 }
 
-function FigureOutBuiltinMediaType($mediatypedvd, $mediatypehddvd, $mediatypebluray) {
+function FigureOutBuiltinMediaType($mediatypedvd, $mediatypehddvd, $mediatypebluray,$mediatypeultrahd) {
```
Nitpicking: 
```suggestion
function FigureOutBuiltinMediaType($mediatypedvd, $mediatypehddvd, $mediatypebluray, $mediatypeultrahd) {
```

gjnoonan replied:
resolved in v2

julmud reviewed (commented):
I still want to do a few tests later today or tomorrow evening before merging, but it looks good. Thanks a lot!

gjnoonan commented:
> I still want to do a few tests later today or tomorrow evening before merging, but it looks good. Thanks a lot!

Thanks @julmud I will re-roll with the above changes today.

gjnoonan committed:
.gitignore: ignore images and debug logs

If you are working on/testing the project and have $getimages set to 2
in your siteconfig it will attempt to grab cover art from the invelos
server, a downside is where it puts them. Let's ignore them.

Also ignore ERROR_LOG and DEBUG_LOG which should never be added into
git.

Signed-off-by: Gavin-John Noonan <mail@gjnoonan.co.uk>

gjnoonan commented:
@julmud I have pushed up V2 of the changes incorporating a suggested change, and introducing a new commit 02b7a5e3e98d4d3eb84ad7ec2b65e73ef0ff01b2 which will add a .gitignore which has your suggested change also.

julmud reviewed (approved):
I've realised tonight that there're a few more files we should add to .gitignore, but I'll leave it for another PR. Thanks a lot!
